subprojects/launcher/src/integTest/groovy/org/gradle/launcher/daemon/server/scaninfo/DaemonScanInfoIntegrationSpec.groovy

```diff
@@ -56,7 +56,8 @@ class DaemonScanInfoIntegrationSpec extends DaemonIntegrationSpec {
     }
 
     //IBM JDK adds a bunch of environment variables that make the foreground daemon not match
-    @Requires(TestPrecondition.NOT_JDK_IBM)
+    //Java 9 and above needs --add-opens to make environment variable mutation work
```

Comment:
The `--foreground` tests have given me lots of headaches before and generally I could never make that option work "in the real world". It's supposed to start a daemon in the client VM, which you can then connect to. But the VM arguments only line up under carefully crafted conditions (i.e. in this test). I think we should deprecate it. Using `tail` on the daemon log and using `org.gradle.debug=true` should provide enough debugging capability.